src/mpags-cipher.cpp

```diff
@@ -86,70 +62,58 @@ int main(int argc, char* argv[])
 
     // Initialise variables
     char inputChar{'x'};
-    std::string inputText;
+    std::string inputText{""};
+    std::string readinText{""}; //string to process text from input file
+    std::string outText{""};
 
     // Read in user input from stdin/file
-    // Warn that input file option not yet implemented
-    if (!inputFile.empty()) {
-        std::cerr << "[warning] input from file ('" << inputFile
-                  << "') not implemented yet, using stdin\n";
-    }
-
-    // loop over each character from user input
-    while (std::cin >> inputChar) {
-        // Uppercase alphabetic characters
-        if (std::isalpha(inputChar)) {
-            inputText += std::toupper(inputChar);
-            continue;
+    if (!inputFile.empty()) {   //if input file is not empty on cmd line arguments
+        std::ifstream in_file{inputFile};   //setup an ifstream of the inputfile name
+        bool readinok = in_file.good(); //setup bool to check if readin is ok
+        if(readinok){   //if readin is true -- i.e. we can readin from the file supplied to the cmd line 
+            in_file >> readinText;  //store input from file in readinText
+            for(std::size_t j{0}; j < readinText.length(); ++j){    //loop over each character in the string
+                inputChar = readinText[j];  //assign the character to inputChar
+                inputText += transformChar(inputChar);  //append to inputText
+            }
+            if(encrypt or decrypt) {    //if encryption or decryption is specified act on this
```

Comment:
Though `or` works, in C++ it's generally more accepted to use the symbol operators, e.g. `&&`, `||`, `==`